packages/foundry/contracts/hooks/rebalancer/Rebalancer.sol

```diff
@@ -0,0 +1,633 @@
+    ) public override onlyVault returns (bool) {
+        ///@dev removing for testing
+        // if (!IBasePoolFactory(weightedPoolFactory).isPoolFromFactory(pool)) {
+        //     revert OnlyWeightedPoolsAllowed();
+        // }
+
+        emit RebalancerHookRegistered(address(this), pool);
+
+        return true;
+    }
+
+    /// @inheritdoc IHooks
+    function getHookFlags() public pure override returns (HookFlags memory) {
+        HookFlags memory hookFlags;
+        hookFlags.shouldCallBeforeAddLiquidity = true;
+        hookFlags.shouldCallAfterRemoveLiquidity = true;
+        hookFlags.shouldCallComputeDynamicSwapFee = true;
+        hookFlags.shouldCallAfterSwap = true;
+        return hookFlags;
+    }
+
+    /// @inheritdoc BaseHooks
+    function onBeforeAddLiquidity(
+        address router,
+        address,
+        AddLiquidityKind,
+        uint256[] memory,
+        uint256,
+        uint256[] memory,
+        bytes memory
+    ) public view override onlySelfRouter(router) returns (bool) {
+        // We only allow addLiquidity via the Router/Hook itself
+        return true;
+    }
+
+    /// @inheritdoc BaseHooks
+    function onAfterRemoveLiquidity(
+        address router,
+        address pool,
+        RemoveLiquidityKind,
+        uint256,
+        uint256[] memory,
+        uint256[] memory amountsOutRaw,
+        uint256[] memory,
+        bytes memory userData
+    ) public override onlySelfRouter(router) returns (bool, uint256[] memory hookAdjustedAmountsOutRaw) {
+        return (true, amountsOutRaw);
+    }
+
+    /// @inheritdoc IHooks
+    function onAfterSwap(
+        AfterSwapParams calldata params
+    ) public override onlyVault returns (bool success, uint256 hookAdjustedAmountCalculatedRaw) {
+        address pool = params.pool;
+        if (rebalanceData[pool].length == 0) {
+            revert RebalanceDataNotSet(pool);
+        }
+
+        (bool rebalanceRequired, uint256[] memory priceActionRatio) = isRebalanceRequired(pool);
+        if (rebalanceRequired) {
+            emit RebalanceStarted(pool);
+            rebalance(pool, priceActionRatio);
+        }
+
+        return (true, params.amountCalculatedRaw);
+    }
+
+    /// @inheritdoc IHooks
+    function onComputeDynamicSwapFeePercentage(
+        PoolSwapParams calldata params,
+        address pool,
+        uint256 staticSwapFeePercentage
+    ) public view override onlyVault returns (bool, uint256) {
+        // uint256 dynamicFee = IOracle(oracle).getFee(pool);
+
+        return (true, staticSwapFeePercentage);
+    }
+
+    /***************************************************************************
+                                  Setter Functions
+    ***************************************************************************/
+    function setWeightedPoolFactoryAddress(address newWeightedPoolFactory) external onlyOwner {
+        ///@dev I am a little confused that whether to check if the factory is disable or not would love
```

Comment:
If the factory is disabled, no more pools can be created from it. Existing pools will continue to work (unless we're migrating because of an issue and they're all paused).

In production, either you'd have a single immutable factory - meaning you'd need to redeploy a new hook with a new factory if Balancer migrated it - or have a set of valid factories that you could add/remove from.

Reply:
Got it thanks for the answering